sdk/storage/ustorageclient/build_all/configs/ustorageclient_build_rules.cmake

```diff
@@ -0,0 +1,144 @@
+        endif()
+        set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS}" PARENT_SCOPE)
+        set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS}" PARENT_SCOPE)
+    endif()
+endfunction()
+
+# Global variable to know if we are on linux, windows, or macosx.
+if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
+    set(ULIB_PAL_DIRECTORY "MSBUILD/X86")
+elseif(CMAKE_SYSTEM_NAME STREQUAL "Linux")
+    # Enable valgrind on Linux
+    find_program(MEMORYCHECK_COMMAND valgrind)
+    set(MEMORYCHECK_COMMAND_OPTIONS "--leak-check=full --error-exitcode=1")
+    set(ULIB_PAL_DIRECTORY "GCC/LINUX")
+elseif(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
+    add_definitions(-DMACOSX)
+    set(ULIB_PAL_DIRECTORY "GCC/IOS")
+endif()
+
+#Enable testing for the target
+enable_testing()
+include(CTest)
+set(use_cppunittest ON)
+
+#Provide stdint and stbool headers if necessary
+if((NOT HAVE_STDINT_H) OR (NOT HAVE_STDBOOL_H))
+    include_directories(${UMOCK_C_INC_FOLDER}/umock_c/aux_inc)
+endif()
+
+# System-specific compiler flags
+if(MSVC)
+    add_definitions(-D_CRT_SECURE_NO_WARNINGS)
+    if(WINCE) # Be lax with WEC 2013 compiler
+        add_definitions(-DWIN32) #WEC 2013
+        # Don't treat warning as errors for WEC 2013. WEC 2013 uses older compiler version
+        add_definitions(/WX-)
+    else()
+        # Make warning as error
+        add_definitions(/WX)
+    endif()
+else()
+    # Make warning as error
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Werror")
+    set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} -Werror")
+    if(NOT(IN_OPENWRT OR APPLE))
+        set(CMAKE_C_FLAGS "-D_POSIX_C_SOURCE=200112L ${CMAKE_C_FLAGS}")
+    endif()
+endif()
+
+#Detect and set target architecture
+include(CheckSymbolExists)
+function(detect_architecture symbol arch)
+    if(NOT DEFINED ARCHITECTURE OR ARCHITECTURE STREQUAL "")
+        set(CMAKE_REQUIRED_QUIET 1)
+        check_symbol_exists("${symbol}" "" ARCHITECTURE_${arch})
+        unset(CMAKE_REQUIRED_QUIET)
+
+        # The output variable needs to be unique across invocations otherwise
+        # CMake's crazy scope rules will keep it defined
+        if(ARCHITECTURE_${arch})
+            set(ARCHITECTURE "${arch}" PARENT_SCOPE)
+            set(ARCHITECTURE_${arch} 1 PARENT_SCOPE)
+            add_definitions(-DARCHITECTURE_${arch}=1)
+        endif()
+    endif()
+endfunction()
+
+if(MSVC)
+    detect_architecture("_M_AMD64" x86_64)
+    detect_architecture("_M_IX86" x86)
+    detect_architecture("_M_ARM" ARM)
+else()
+    detect_architecture("__x86_64__" x86_64)
+    detect_architecture("__i386__" x86)
+    detect_architecture("__arm__" ARM)
+endif()
+if (NOT DEFINED ARCHITECTURE OR ARCHITECTURE STREQUAL "")
+    set(ARCHITECTURE "GENERIC")
+endif()
+message(STATUS "target architecture: ${ARCHITECTURE}")
+
+#if any compiler has a command line switch called "OFF" then it will need special care
+if(NOT "${compileOption_C}" STREQUAL "OFF")
+    set(CMAKE_C_FLAGS "${compileOption_C} ${CMAKE_C_FLAGS}")
```

Comment:
this is a really bizarre use of options, just have the "OFF" value be an empty string and always append it, or better yet don't bother with this option and just have users add to CMAKE_C_FLAGS in their cache, or by settting the CFLAGS env var.